vt/client.py

```diff
@@ -973,3 +975,59 @@ async def _wait_for_analysis_completion(self, analysis: Object) -> Object:
 
   async def wait_for_analysis_completion(self, analysis: Object) -> Object:
     return await self._wait_for_analysis_completion(analysis)
+
+
+  def scan_file_private(
+      self, 
+      file: typing.Union[typing.BinaryIO, str],
+      code_insight: bool = False,
+      wait_for_completion: bool = False
+  ) -> Object:
+      """Scan file privately with optional code insight analysis.
+      
+      Args:
+          file: File to scan (path string or file object)
+          code_insight: Enable code analysis features
+          wait_for_completion: Wait for completion
+          
+      Returns:
+          Object: Analysis object with scan results
+      """
+      return make_sync(
+          self.scan_file_private_async(file, code_insight, wait_for_completion)
+      )
+
+  async def scan_file_private_async(
+      self,
+      file: typing.Union[typing.BinaryIO, str],
+      code_insight: bool = False, 
+      wait_for_completion: bool = False
+  ) -> Object:
+      """Async version of scan_file_private"""
+
+      # Handle string path
+      if isinstance(file, str):
+          async with aiofiles.open(file, 'rb') as f:
+              file_content = io.BytesIO(await f.read())
+              file_content.name = os.path.basename(file)
+              return await self.scan_file_private_async(
+                  file_content,
+                  code_insight=code_insight,
+                  wait_for_completion=wait_for_completion
+              )
+
+      # Create form data for private scan
+      form = aiohttp.FormData()
+      form.add_field('file', file)
+      form.add_field('code_insight', '1' if code_insight else '0')
```

Comment:
I'm still hesitant about this parameter here, where did you see documentation for this specific parameter?

Reply:
All available parameters are defined here: https://docs.virustotal.com/reference/upload-file-private-scanning

The code insight one is not listed.

A group can disable AI features (such as code insights) but it must be done patching the group settings and only by a group administrator.